pkg/addons/addons.go

```diff
@@ -200,6 +208,9 @@ func writeDB(client Interface, db map[string]unstructured.Unstructured) error {
 	if err := client.ApplyResources(cfgFilter, db, keys); err != nil {
 		return err
 	}
+
+	//TODO: validate if CRD and SC is available here.
```

Comment:
I think we should replace `client.ServiceCatalogExists` below with something more generic to check SC and CRD objects (or even second method). and check here, and only then they are available to move forward. We will need this as in example `Istio` will have around 28 CRDs... We will see more of this coming.

Reply:
Separate object better. APi server takes ages to aggregate as it needs infra nodes available. CRD on other hand is almost instant.